common/lib/xmodule/xmodule/modulestore/search.py

```diff
@@ -104,18 +105,29 @@ def find_path_to_course():
         # (e.g. sequential and videosequence) currently deal with this form of
         # representing nested positions. This needs to happen before jumping to a
         # module nested in more than one positional module will work.
+
         if n > 3:
             position_list = []
+            masquerading_as_student = False
+            if request and request.session.get('masquerade_settings') and \
+                    request.session.get('masquerade_settings')[usage_key.course_key].role == 'student':
+                masquerading_as_student = True
+            include_staff_only_blocks = request and GlobalStaff().has_user(request.user) and not masquerading_as_student
             for path_index in range(2, n - 1):
                 category = path[path_index].block_type
                 if category == 'sequential' or category == 'videosequence':
                     section_desc = modulestore.get_item(path[path_index])
                     # this calls get_children rather than just children b/c old mongo includes private children
                     # in children but not in get_children
-                    child_locs = [c.location for c in section_desc.get_children()]
+                    if not include_staff_only_blocks:
+                        child_locs = [c.location for c in section_desc.get_children()
+                                      if c.visible_to_staff_only is False]
```

Comment:
can we have this statement in a single line for more readability? or break it to multiple lines?